homeassistant/components/sensor/broadlink.py

```diff
@@ -0,0 +1,327 @@
+            try:
+                packet = bytearray(0x38)
+                packet[0x00] = 0x5a
+                packet[0x01] = 0xa5
+                packet[0x02] = 0xaa
+                packet[0x03] = 0x55
+                packet[0x04] = 0x5a
+                packet[0x05] = 0xa5
+                packet[0x06] = 0xaa
+                packet[0x07] = 0x55
+                packet[0x24] = 0x2a
+                packet[0x25] = 0x27
+                packet[0x26] = command
+                packet[0x28] = self.count & 0xff
+                packet[0x29] = self.count >> 8
+                packet[0x2a] = self.mac[0]
+                packet[0x2b] = self.mac[1]
+                packet[0x2c] = self.mac[2]
+                packet[0x2d] = self.mac[3]
+                packet[0x2e] = self.mac[4]
+                packet[0x2f] = self.mac[5]
+                packet[0x30] = self.ip_arr[0]
+                packet[0x31] = self.ip_arr[1]
+                packet[0x32] = self.ip_arr[2]
+                packet[0x33] = self.ip_arr[3]
+            except (IndexError, TypeError, NameError):
+                _LOGGER.error('Invalid IP or MAC address.')
+                return bytearray(0x30)
+
+            checksum = 0xbeaf
+            for i, item in enumerate(payload):
+                checksum += payload[i]
+                checksum = checksum & 0xffff
+
+            aes = AES.new(bytes(self.key), AES.MODE_CBC, bytes(self.ivr))
+            payload = aes.encrypt(bytes(payload))
+
+            packet[0x34] = checksum & 0xff
+            packet[0x35] = checksum >> 8
+
+            for i, item in enumerate(payload):
+                packet.append(payload[i])
+
+            checksum = 0xbeaf
+            for i, item in enumerate(packet):
+                checksum += packet[i]
+                checksum = checksum & 0xffff
+            packet[0x20] = checksum & 0xff
+            packet[0x21] = checksum >> 8
+
+            with self.lock:
+                self.sock.sendto(packet, self.host)
+                try:
+                    self.sock.settimeout(timeout)
+                    response = self.sock.recvfrom(1024)
+                except socket.timeout:
+                    _LOGGER.error("Socket timeout...")
+                    return bytearray(0x30)
+
+            return response[0]
+
+        def check_sensors(self):
+            """Collect data from Broadlink sensors."""
+            packet = bytearray(16)
+            packet[0] = 1
+            response = self.send_packet(0x6a, packet)
+            err = response[0x22] | (response[0x23] << 8)
+            if err == 0:
+                data = {}
+                aes = AES.new(bytes(self.key), AES.MODE_CBC, bytes(self.ivr))
+                payload = aes.decrypt(bytes(response[0x38:]))
+                data['temperature'] = (payload[0x4] * 10 + payload[0x5]) / 10.0
+                data['humidity'] = (payload[0x6] * 10 + payload[0x7]) / 10.0
+                light = payload[0x8]
+                if light == 0:
```

Comment:
I would prefer numbers instead of text variables. That will give nice graphs in HA to track the change